### ISSUE TITLE
[v469c] "You killed..." message doesn't appear after depressurizing an opponent in DM-Pressure.

### ISSUE THREAD

**Issue by NeonKnightOA** [OldUnreal/UnrealTournamentPatches#833]
Steps for reproduction:

* Play in DM-Pressure against bots.
* Get a bot into the pressure chamber.
* Press one of the lateral buttons and wait until it explodes.

Expected result: "You killed [botname]" message appears in the center of the screen.
Actual result: Nothing happens.

It should be noted that the top-left "X was depressurized by Y" message do appear.

**Comment by SeriousBuggie** [comment]
Faster way of reproduce:
1. Start map without bots: [DM-Pressure_dbg.zip](https://github.com/OldUnreal/UnrealTournamentPatches/files/8934450/DM-Pressure_dbg.zip)
2. Go outside near button.
3. Write in console `addbots 1`
4. Press button.

**Comment by SeriousBuggie** [comment]
PressureZone kill with `SpecialDamage` damage type:
```js
			if ( ratio == 1.0 )
			{	
				Level.Game.SpecialDamageString = DamageString;
				P.health = -1000; //make sure gibs
				P.Died(Instigator, 'SpecialDamage', P.Location);
				MakeNormal(P);
			}
```
so later in GameInfo.Killed:
```js
		if ( (DamageType == 'SpecialDamage') && (SpecialDamageString != "") )
		{
			BroadcastMessage( ParseKillMessage(
					Killer.PlayerReplicationInfo.PlayerName,
					Other.PlayerReplicationInfo.PlayerName,
					Killer.Weapon.ItemName,
					SpecialDamageString
					),
				false, 'DeathMessage');
			bSpecialDamage = True;
		}
```
and regular death message not called:
```js
				if (!bSpecialDamage && (Other != None))
				{
					BroadcastRegularDeathMessage(Killer, Other, damageType);
				}
```
Which is that one:
```js
function BroadcastRegularDeathMessage(pawn Killer, pawn Other, name damageType)
{
	BroadcastLocalizedMessage(DeathMessageClass, 0, Killer.PlayerReplicationInfo, Other.PlayerReplicationInfo, Killer.Weapon.Class);
}
```
Default props from PressureZone:
```js
defaultproperties
{
	DamageType="SpecialDamage"
	DamageString="%o was depressurized by %k."
	bStatic=False
}
```
DeathMatchPlus use next DeathMessageClass for usual kill:
`DeathMessageClass=Class'Botpack.DeathMessagePlus'`
When you pressure it goes as `'DeathMessage'` according to call above.
In ChallengeHUD it goes as:
```js
		case 'DeathMessage':
			MessageClass = class'RedSayMessagePlus';
			break;
```
Which is not same as `DeathMessagePlus`, but more like info message.
![img](https://user-images.githubusercontent.com/70026933/174459250-f420956a-8043-4f38-8ede-dce35cb5fba8.png)
On ClientReceive, DeathMessagePlus born two messages:
1. Self,
2. Child class.
```js
		// Interdict and send the child message instead.
		if ( TournamentPlayer(P).myHUD != None )
		{
			TournamentPlayer(P).myHUD.LocalizedMessage( Default.ChildMessage, Switch, RelatedPRI_1, RelatedPRI_2, OptionalObject );
			TournamentPlayer(P).myHUD.LocalizedMessage( Default.Class, Switch, RelatedPRI_1, RelatedPRI_2, OptionalObject );
		}
```
props:
```js
	KilledString="was killed by"
	ChildMessage=Class'Botpack.KillerMessagePlus'
```
Which lead to KillerMessagePlus:
```js
	YouKilled="You killed"
```
```js
	if (RelatedPRI_1 == None)
		return "";
	if (RelatedPRI_2 == None)
		return "";

	if (RelatedPRI_2.PlayerName != "")
		return Default.YouKilled@RelatedPRI_2.PlayerName@Default.YouKilledTrailer;
```
So there appear your kill message.

TL;DR
PressureZone use own type of death, which dispatch differently and not produce kill messages as regular death.

**Comment by SeriousBuggie** [comment]
AFAIS there nothing to "fix". Need implement additional DeatchMesagePlus subclass in additional package, so old clients can load it and use. After that need alter PressureZone for send this special type of message on kill. After that there be Killed message + old depressured message.
